r2/r2/lib/base.py

```diff
@@ -104,7 +107,7 @@ def __call__(self, environ, start_response):
             request.ip = cdn_ip
             request.via_cdn = True
         elif g.trust_local_proxies and forwarded_for and is_local_address(remote_addr):
-            request.ip = forwarded_for.split(',')[-1]
+            request.ip = forwarded_for.split(',')[0].strip()
```

Comment:
> This reads from the opposite end of the list of forwarded-fors, is that intended?

It is.  According to:

http://docs.aws.amazon.com/ElasticLoadBalancing/latest/DeveloperGuide/x-forwarded-headers.html#x-forwarded-for and
https://www.citrix.com/blogs/2012/09/27/multiple-xff-ips/

the X-Forwarded-For header takes the form:

> X-Forwarded-For: OriginatingClientIPAddress, proxy1-IPAddress, proxy2-IPAddress

This is further corroborated by the behaviour of Varnish cache wrt the XFF header.

Since `request.ip` is supposed to be the requesting client's IP, not that of any of the intermediate proxies, the line in question should be taking the first IP in the list, not the last.

With standard HAProxy installations, this bug would probably never surface because there would never be more than one IP in XFF.  But, in my Varnish-based install, SSL connections are first terminated by nginx, which proxies to Varnish, which proxies on to the paste reddit process so XFF contains two IPs and, as currently written, reddit records the wrong IP.

God knows how the business end of production reddit goes in respect of proxies, caches and load-balancers.  As a computer scientist and electronic engineer by profession, I'd love to know, but I guess that probably comes under the heading of 'trade secret'.

Without the strip() call, [errant whitespace causes memcached exceptions](http://pastebin.com/1QQiRsgs).  Per the PR text, I instrumented the code (with a little guidance from bsimpson) and the culprit key was (note the leading whitespace in the string):
`[' 127.0.0.1']`